modules/programs/khal.nix

```diff
@@ -0,0 +1,55 @@
+# khal config loader is sensitive to leading space !
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.khal;
+
+  khalAccounts = filterAttrs (_: a: a.khal.enable)
+    (config.accounts.calendar.accounts);
+in
+
+{
+  options.programs.khal = {
+    enable = mkEnableOption "khal, a CLI calendar application";
+  };
+
+  config = mkIf cfg.enable {
+    home.packages =  [ pkgs.khal ];
+
+    xdg.configFile."khal/config".text = concatStringsSep "\n" (
+    [
+      "[calendars]"
+    ]
+    ++ (mapAttrsToList (name: value: concatStringsSep "\n"
+      ([
+        ''[[${name}]]''
+        ''path = ${value.vdirsyncer.local.path + (if value.khal.type == "discover" then "/*" else "/")}''
```

Comment:
I think the fact that you need to access the `vdirsyncer` configuration here suggests that the local calendar configuration should be moved to the account directly, at least the parts that don’t immediately reference `vdirsyncer`.

Reply:
Very true. I was trying to keep in line with what you had already done so I've moved it out now.